### DISCUSSION TITLE
Reorder columns

### DISCUSSION THREAD

**Discussion by nikita-galileo** [huggingface/datasets#4646] [Q&A] [answered]
Is there a way to reorder the columns in a dataset? I notice `remove_columns` and `rename_columns` and have even tried the following to no avail:
```
def reorder_cols(sample):
    sample = {col: sample[col] for col in new_cols_order}
    return sample

new_ds = new_ds.map(reorder_cols)
```

**Comment by mariosasko** [comment, marked as answer]
Hi! The columns can be reordered either by performing a `Dataset.cast` with the reordered features dictionary or by replacing the initial features with the reordered ones (ds.info.features = reordered_features). I suggest you use the former approach as the latter relies on the internal decoding logic.
